lib/_emerge/depgraph.py

```diff
@@ -5250,7 +5250,13 @@ def _resolve(self, myfavorites):
         # set below is reserved for cases where there are *zero* other
         # problems. For reference, see backtrack_depgraph, where it skips the
         # get_best_run() call when success_without_autounmask is True.
-        if self._have_autounmask_changes():
+
+        if onlydeps and self._dynamic_config._needed_license_changes:
+            # needed license changes should only be fatal for packages that
+            # would actually be installed
+            return True, myfavorites
```

Comment:
You've ignored non-license autounmask changes here. It would only make sense to return True here if there were no non-license autounmask changes. Ultimately, I think it will be much cleaner to set empty LICENSE metadata for the onlydeps package instance, rather than trigger autoumask for a LICENSE that you intend to ignore anyway.

Reply:
Setting license metadata to empty is an excellent suggestion. I'll look into it. Thanks!